lib/rspec/core/world.rb

```diff
@@ -19,6 +19,17 @@ def initialize(configuration=RSpec.configuration)
         end
       end
 
+      # Invokes block before registering an example group
+      def before_register(&block)
+        before_register_callbacks << block
+      end
```

Comment:
I think this should go on config instead of world.  Config is where we let users configure callbacks.

Also, example group registration isn't really a concept we've exposed into our API up to now.  It's purely an internal.  I think that this feature would fit better with RSpec's terminology if it was something like `on_example_group_definition`, e.g.:

``` ruby
RSpec.configure do |c|
  c.on_example_group_definition do |example_group|
    # do something with the newly defined example group
  end
end
```